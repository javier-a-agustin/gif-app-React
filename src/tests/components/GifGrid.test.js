import React from 'react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
const { shallow } = require("enzyme");
const { GifGrid } = require("../../components/GifGrid");
jest.mock('../../hooks/useFetchGifs');

describe('', () => {
    const category = 'Transformers';

    test('debe mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category }/>);

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar los items cuando se cargan las imagenes de useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/asda.jpg',
            title: 'Cualquiera',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category={ category }/>);

        expect( wrapper ).toMatchSnapshot();

        expect( wrapper.find('p').exists() ).toBe(false);

        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });



});