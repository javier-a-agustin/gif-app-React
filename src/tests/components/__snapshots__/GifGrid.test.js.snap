// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
     
    Transformers
     
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    ;
  </div>
</Fragment>
`;

exports[`debe mostrar los items cuando se cargan las imagenes de useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    Transformers
     
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquiera"
      url="https://localhost/asda.jpg"
    />
    ;
  </div>
</Fragment>
`;
